frontend/packages/pipelines-plugin/src/components/repository/RepositoryDetails.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { GithubIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import { SectionHeading, ResourceSummary, ExternalLink } from '@console/internal/components/utils';
+import { RepositoryKind } from './types';
+
+export interface RepositoryDetailsProps {
+  obj: RepositoryKind;
+}
+
+const RepositoryDetails: React.FC<RepositoryDetailsProps> = ({ obj: repository }) => {
+  const { t } = useTranslation();
+  const { spec } = repository;
+
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text={t('pipelines-plugin~Repository details')} />
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={repository} />
+        </div>
+        <div className="col-sm-6">
+          <dl>
+            <dt>{t('pipelines-plugin~Repository')}</dt>
+            <dd>
+              <ExternalLink href={spec.url}>
+                <GithubIcon title={spec.url} /> {spec.url}
+              </ExternalLink>
+            </dd>
+            {spec.branch && (
```

Comment:
PAC Repository CR doesn't have the real validations for the `spec` field as of now. Technically its possible for us to remove the `spec` field today and save the Repository CR. So until there is a real validation for spec, we should treat `spec` field as optional. 

Updated the Repository CR types and add optional chaining here as well.


Ref: https://coreos.slack.com/archives/C01T16K6C4A/p1628496190002100

Reply:
Can we add unit test cases for this file ?

Reply:
added unit test and updated